bookstore-scrapper/algolia/updateAlgoliaIndex.ts

```diff
@@ -39,33 +46,32 @@ const rankings: string[] = [
 ];
 
 export const getAlgoliaObject = async (): Promise<AlgoliaEntry[]> => {
-  const dbBooks: Book[] = await prisma.book.findMany();
+  const dbBooks: BookWithUserWithCourseWithDept[] = await prisma.book.findMany({
+    include: {
+      courses: {
+        include: {
+          dept: true,
+        },
+      },
+    },
+  });
 
   const dbCourses: Course[] = await prisma.course.findMany({
     include: {
       dept: true,
     },
   });
 
-  const algoliaBookEntries: AlgoliaEntry[] = dbBooks
-    .filter((book) => {
-      return (
-        /* Remove campus store entries that are not related to physical books
-        or provide a note to users. */
-        !(
-          book.isbn.endsWith(UNWANTED_ENTRY_KEYWORD.ISBN_ENDING_WITH_B) ||
-          book.name.startsWith(UNWANTED_ENTRY_KEYWORD.ETEXT) ||
-          book.name.includes(UNWANTED_ENTRY_KEYWORD.LAB_MANUAL)
-        )
-      );
-    })
-    .map((book: Book) => {
-      const algoliaBookEntry: AlgoliaEntry = {
-        type: EntryType.BOOK,
-        entry: book,
-      };
-      return algoliaBookEntry;
-    });
+  const algoliaBookEntries: AlgoliaEntry[] = [];
+
+  for (const book of dbBooks) {
+    const googleBooksData = await getGoogleBooksData(book.isbn);
```

Comment:
It looks like you don't re-request when the data wasn't found due to the request limit. Is it worthwhile to distinguish between "Too many requests" and "Not found", so that you can handle this somehow when updating the index?

Reply:
Using API Key moving forwards. Resolved.

Reply:
Just to confirm, the search just returns an empty array if nothing was found. 404 error is not returned.